Utility/Connect-AzureVM.ps1

```diff
@@ -27,103 +27,198 @@
+	subscription, gets the public IP Address of the virtual machine and return it. 
 
-.PARAMETER AzureSubscriptionName
-    Name of the Azure subscription to connect to.
+.PARAMETER AzureSubscriptionId
+    SubscriptionId of the Azure subscription to connect to
     
-.PARAMETER AzureOrgIdCredential
+.PARAMETER AzureOrgIdCredentialName
     A credential containing an Org Id username / password with access to this Azure subscription.
 
-	If invoking this runbook inline from within another runbook, pass a PSCredential for this parameter.
+	If invoking this runbook inline from within another runbook, pass a Azure Automation PSCredential Name for this parameter.
 
-	If starting this runbook using Start-AzureAutomationRunbook, or via the Azure portal UI, pass as a string the
-	name of an Azure Automation PSCredential asset instead. Azure Automation will automatically grab the asset with
-	that name and pass it into the runbook.
-
-.PARAMETER ServiceName
-    Name of the cloud service where the VM is located.
+.PARAMETER ResourceGroupName
+    Name of the resource group where the VM is located.
 
 .PARAMETER VMName    
-    Name of the virtual machine that you want to connect to. 
+    Name of the virtual machine that you want to connect to  
 
 .EXAMPLE
-    Connect-AzureVM -AzureSubscriptionName "Visual Studio Ultimate with MSDN" -ServiceName "Finance" -VMName "WebServer01" -AzureOrgIdCredential $cred
+    Connect-AzureARMVMPS -AzureSubscriptionId "1019**********************" -ResourceGroupName "RG1" -VMName "VM01" -AzureOrgIdCredentialName $cred
 
 .NOTES
-    AUTHOR: System Center Automation Team
-    LASTEDIT: Dec 18, 2014 
+    AUTHOR: Rohit Minni
+    LASTEDIT: June 29, 2016 
 #>
-workflow Connect-AzureVM
-{
-	[OutputType([System.Uri])]
-
+   
     Param
     (            
         [parameter(Mandatory=$true)]
-        [String]
-        $AzureSubscriptionName,
+        [String]$AzureSubscriptionId,
 
-		[parameter(Mandatory=$true)]
-        [PSCredential]
-        $AzureOrgIdCredential,
-        
         [parameter(Mandatory=$true)]
-        [String]
-        $ServiceName,
+        [String]$AzureOrgIdCredentialName,
+                        
+        [parameter(Mandatory=$true)]
+        [String]$ResourceGroupName,
         
         [parameter(Mandatory=$true)]
-        [String]
-        $VMName      
+        [String]$VMName      
     )
-   
-    Add-AzureAccount -Credential $AzureOrgIdCredential | Write-Verbose
+
+    $ErrorActionPreference = "SilentlyContinue"
+    $AzureOrgIdCredential = Get-AutomationPSCredential -Name $AzureOrgIdCredentialName
+	
+    $Login = Login-AzureRmAccount -Credential $AzureOrgIdCredential
 
 	# Select the Azure subscription we will be working against
-    Select-AzureSubscription -SubscriptionName $AzureSubscriptionName | Write-Verbose
+    $Subscription = Select-AzureRmSubscription -SubscriptionId $AzureSubscriptionId
+
+
+  function Configure-AzureWinRMHTTPS {
```

Comment:
why have this in its own function if the function is only called once? adds more lines for not much value...